client/state/site-settings/exporter/actions.js

```diff
@@ -26,41 +43,168 @@ export function setPostType( postType ) {
 }
 
 /**
- * Sends a request to the server to start an export.
+ * Sets one of the advanced export settings values in the UI
  *
+ * @param  {string} section   The name of the section containing the setting - 'posts', 'pages', or 'feedback'
+ * @param  {string} setting   The name of the setting
+ * @param  {any}    value     The new value for the setting
+ * @return {Object}           Action object
+ */
+export function setAdvancedSetting( section, setting, value ) {
+	return {
+		type: SET_EXPORTER_ADVANCED_SETTING,
+		section,
+		setting,
+		value
+	};
+}
+
+/**
+ * Ensure that we are fetching or have fetched the available export
+ * settings for the current site.
+ *
+ * @param  {int}      siteId  The ID of the site for which to retrieve export settings
  * @return {Function}         Action thunk
  */
-export function startExport() {
+export function ensureHasSettings() {
+	return ( dispatch, getState ) => {
+		const state = getState();
+		const selectedSite = getSelectedSite( state );
+		const currentDataSiteId = getDataState( state ).forSiteId;
+
+		if ( selectedSite ) {
+			if ( selectedSite.ID !== currentDataSiteId ) {
+				dispatch( fetchExportSettings( selectedSite.ID ) );
+			}
+		}
+	}
+}
+
+/**
+ * Request the available settings for customizing an export.
+ *
+ * @param  {int}      siteId  The ID of the site for which to retrieve export settings
+ * @return {Function}         Action thunk
+ */
+export function fetchExportSettings( siteId ) {
 	return ( dispatch ) => {
 		dispatch( {
-			type: REQUEST_START_EXPORT
+			type: FETCH_EXPORTER_ADVANCED_SETTINGS,
+			siteId: siteId
+		} );
+
+		wpcomUndocumented.getExportSettings( siteId, ( error, data ) => {
+			dispatch( receiveExportSettings( siteId, data ) );
+		} );
```

Comment:
Hearing back to my earlier comment in `Undocumented`, this could be written as follows if we used the promises from `wpcom.js`

```js
wpcomUndocumented
	.getExportSettings( siteId )
	.then( data => dispatch( receiveExportSettings( siteId, data ) ) );
```

Or with the addition of a small helper function, we can write it in _point-free_ style, which I personally find more readable

```js
// probably too crazy
const updateExportSettings = compose( dispatch, partial( receiveExportSettings, siteId ) );

// less-crazy helper function
const updateExportSettings = settings => dispatch( receiveExportSettings( siteId, settings ) );

wpcomUndocumented
	.getExportSettings( siteId )
	.then( updateExportSettings );
```